engine.py

```diff
@@ -161,114 +160,72 @@ def log_error(self, msg):
         self._log.error(msg)
 
     ##########################################################################################
-    # pyside / qt
-    
-    def _define_qt_base(self):
-        """
-        check for pyside then pyqt
-        """
+    # qt
+
+    def _define_unavailable_base(self):
         # proxy class used when QT does not exist on the system.
         # this will raise an exception when any QT code tries to use it
-        class QTProxy(object):                        
+        class QTProxy(object):
             def __getattr__(self, name):
                 raise tank.TankError("Looks like you are trying to run an App that uses a QT "
                                      "based UI, however the Shell engine could not find a PyQt "
-                                     "or PySide installation in your python system path. We " 
+                                     "or PySide installation in your python system path. We "
                                      "recommend that you install PySide if you want to "
                                      "run UI applications from the Shell.")
+        return {
+            "qt_core": QTProxy(),
+            "qt_gui": QTProxy(),
+            "qt_widgets": QTProxy(),
+            "dialog_base": None
+        }
+
+    def _define_qt_base(self):
+        """
+        check for pyside then pyqt
+        """
+        base = Engine._define_qt_base(self)
+
+        # If QtCore hasn't been set, then nothing was.
+        if base["qt_core"] is None:
+            return self._define_unavailable_base()
+
+        try:
+            QtCore = base["qt_core"]
+            DialogBase = base["dialog_base"]
+            QtWrapper = base["wrapper"]
```

Comment:
Is this wrapper key new? Should we fail gracefully if it's not found (e.g. if user is running a current core)? Or set min core constraint for engine?

Reply:
This was introduced in v0.18.8.  Updated the min core.